### PR TITLE
python312Packages.latexrestricted: init at 0.4.0; latexminted: 0.1.0b9 -> 0.1.0b16

### PR DESCRIPTION
**Pull request by romildo**
## Description of changes

- python312Packages.latexrestricted: init at [0.4.0](https://pypi.org/project/latexrestricted/)
- latexminted: 0.1.0b9 -> [0.1.0b16](https://pypi.org/project/latexminted/#description)

`latexrestricted` is a Python library for creating executables compatible with LaTeX restricted shell escape.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/latexrestricted/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-AMfDTruJKejDdXW98VkeeEwELql5566bsL1SutLDpso=";
+  };
+
+  build-system = [
+    setuptools
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/latexrestricted/default.nix:20]: Looks like that you need to run `nixfmt` again.

File: pkgs/development/python-modules/latexrestricted/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = {
+    homepage = "https://github.com/gpoore/latexrestricted";
+    description = "Python library for creating executables compatible with LaTeX restricted shell escape";
```
Comment (fabaff) [inline pkgs/development/python-modules/latexrestricted/default.nix:24]:
```suggestion
    description = "Python library for creating executables compatible with LaTeX restricted shell escape";
    changelog = "https://github.com/gpoore/latexrestricted/blob/v${version}/CHANGELOG.md";
```

File: pkgs/development/python-modules/latexrestricted/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    license = lib.licenses.lppl13c;
+    platforms = lib.platforms.unix;
```
Comment (fabaff) [inline pkgs/development/python-modules/latexrestricted/default.nix:26]: No needed, see https://github.com/gpoore/latexrestricted/blob/main/pyproject.toml#L29

```suggestion
```

File: pkgs/development/python-modules/latexrestricted/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/development/python-modules/latexrestricted/default.nix:20]: I have run the following command, and it did not produced any further changes, and the automatic tests passed.

```console
nix-shell -p nixfmt-rfc-style --run "nixfmt pkgs/development/python-modules/latexrestricted/default.nix"
````

File: pkgs/development/python-modules/latexrestricted/default.nix (hunk 2 above)
Comment (romildo) [reply pkgs/development/python-modules/latexrestricted/default.nix:24]: Done.

File: pkgs/development/python-modules/latexrestricted/default.nix (hunk 3 above)
Comment (romildo) [reply pkgs/development/python-modules/latexrestricted/default.nix]: Done.

File: pkgs/development/python-modules/latexrestricted/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-AMfDTruJKejDdXW98VkeeEwELql5566bsL1SutLDpso=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/latexrestricted/default.nix:21]:
```suggestion
  pythonImportsCheck = [ "latexrestricted" ];
  
  # upstream has no tests
  doCheck = false;
```
Could you add `pythonImportsCheck` for a smoke check?
Also, we encourage enabling tests, so we'd like to explicitly disable `checkPhase` if there is no tests.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/development/python-modules/latexrestricted/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/development/python-modules/latexrestricted/default.nix]: Done.

File: pkgs/development/python-modules/latexrestricted/default.nix (hunk 4 above)
Comment (romildo) [reply pkgs/development/python-modules/latexrestricted/default.nix:24]: Done.

**Pull request merged**